backend/core/src/application-flagged-sets/application-flagged-sets.service.ts

```diff
@@ -115,6 +121,46 @@ export class ApplicationFlaggedSetsService {
     }
   }
 
+  async onApplicationUpdate(
+    newApplication: Application,
+    transactionalEntityManager: EntityManager
+  ) {
+    const transApplicationsRepository = transactionalEntityManager.getRepository(Application)
+    newApplication.markedAsDuplicate = false
+    await transApplicationsRepository.save(newApplication)
+
+    const transAfsRepository = transactionalEntityManager.getRepository(ApplicationFlaggedSet)
+    let afses = await transAfsRepository
+      .createQueryBuilder("afs")
+      .leftJoinAndSelect("afs.applications", "applications")
+      .leftJoinAndSelect("afs.listing", "listing")
+      .where("listing.id = :listingId", { listingId: newApplication.listing.id })
+      .getMany()
+    afses = afses.filter(
+      (afs) => afs.applications.findIndex((app) => app.id === newApplication.id) !== -1
+    )
+    const afsesToBeSaved: Array<ApplicationFlaggedSet> = []
+    const afsesToBeRemoved: Array<ApplicationFlaggedSet> = []
+    for (const afs of afses) {
+      afs.status = FlaggedSetStatus.flagged
+      afs.resolvedTime = null
+      afs.resolvingUser = null
+      const applicationIndex = afs.applications.findIndex(
+        (application) => application.id === newApplication.id
+      )
+      afs.applications.splice(applicationIndex, 1)
+      if (afs.applications.length > 1) {
+        afsesToBeSaved.push(afs)
+      } else {
+        afsesToBeRemoved.push(afs)
+      }
+    }
```

Comment:
We're removing newApplication from the flagged and if there's one still in the flagged set, we're removing that one from the set, leaving only newApplication in the flagged set. Is that what we want?

Reply:
This part of the code is reverting any AFS logic for given application id (as if it was never added to any AFS) and it's also removing logically incorrect AFSes. Once AFS containing application with given appId is found, the application is removed from it and since AFS with just 1 application left does not make sense, it's removed too. It will probably be recreated again by onApplicationSave logic but I just wanted it to be trivial and reuse what already works.